### PR TITLE
Implement parseAssignmentStatements

### LINKED ISSUE #78: [parser] implement parseAssignmentStatement
```
// TODO: implement me w/ test cases :-)
func parseAssignStatement(buf TokenBuffer) (*ast.AssignStatement, []error) {
	return nil, nil
}
```

### PR DESCRIPTION
**Pull request by zeroFruit**
resolved: #78 

DONE!

- [x] Test case

### CODE REVIEW

File: parse/parser.go [hunk 1]
```diff
@@ -280,10 +280,39 @@ func parseReturnStatement(buf TokenBuffer) (ast.Expression, []error) {
 
 // TODO: implement me w/ test cases :-)
 func parseAssignStatement(buf TokenBuffer) (*ast.AssignStatement, []error) {
-	return nil, nil
+	stmt := &ast.AssignStatement{}
+
+	tok := buf.Read()
+	if !isDataStructure(tok) {
```
Comment (boohyunsik) [inline parse/parser.go:380]: This is just a question.
Can koa make user-defined data structure? (like struct or class?)

Comment (zeroFruit) [reply parse/parser.go:380]: yes but we're currently planning to make string, bool, int data structure type

Comment (junbeomlee) [reply parse/parser.go:380]: We will!

Comment (boohyunsik) [reply parse/parser.go:380]: Cool :-D

File: parse/parser.go [hunk 2]
```diff
@@ -350,10 +350,48 @@ func parseReturnStatement(buf TokenBuffer) (ast.Statement, []error) {
...
+	if tok.Type != Ident {
+		return nil, []error{parseError{tok.Type, "token is not identifier"}}
+	}
+
+	stmt.Variable = ast.Identifier{
+		Value: tok.String(),
+	}
+
+	if buf.Read().Type != Assign {
+		return nil, []error{parseError{tok.Type, "token is not assign"}}
```
Comment (junbeomlee) [inline parse/parser.go:399]: how about using `expectNext`

Comment (zeroFruit) [reply parse/parser.go:399]: good!

Comment (zeroFruit) [reply parse/parser.go:399]: no on second thought.. we should move on to next token and check the type

**Comment by zeroFruit** [comment]
\+ little refactor on `expectNext`
\+ resolve conflict

**Review by junbeomlee — APPROVED** [review]

**Comment by junbeomlee** [comment]
LGTM

**Pull request merged**